fix: evaluate session state before FORM rendering and submission

leleueri opened the pull request:
a new attribute has been added to the session to determine if the session is ongoing or not

 if it is actions related to a form are executed as usual

 if it is not a redirect to authorization endpoint is perform to follow the auth flow

 this is to avoid issues when a user is fully authenticated but use the back arraow on the browser

 this solution is not perfect as a back click before the end of the auth flow

 will not skip the page rendering

fixes AM-4342

lgw-gravitee commented on gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/BypassDirectRequestHandler.java:
```diff
@@ -0,0 +1,62 @@
+        if (!skipHandlerExecution && (routingContext.session() == null ||
+                !hasText(routingContext.session().get(ConstantKeys.SESSION_KEY_AUTH_FLOW_STATE)) ||
+                !ConstantKeys.SESSION_KEY_AUTH_FLOW_STATE_ONGOING.equalsIgnoreCase(routingContext.session().get(ConstantKeys.SESSION_KEY_AUTH_FLOW_STATE)))) {
+            // SESSION_KEY_AUTH_FLOW_STATE is used by the authorization endpoint to flag an ongoing authentication
```
It's a little bit blurry. How about extracting part of it into something like `checkAuthFlowState(ctx)`?

lgw-gravitee commented on gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/scope/impl/ScopeManagerImpl.java:
```diff
@@ -75,7 +79,7 @@ public void afterPropertiesSet() {
         eventManager.subscribeForEvents(this, ScopeEvent.class, domain.getId());
-
+        this.alwaysProvideEnhancedScopes = OIDC_ALWAYS_ENHANCE_SCOPE.from(environment);
     }
```
**question** What is the reason behind replacing @Value with Environment? Here, I believe we're adding another dependency?

leleueri replied:
My point was to centralized the "legacy" setings to easily found them. Environment and @Value are both Spring element so not really new dep (we are still dependant to spring)

leleueri committed:
fix: evaluate session state before FORM rendering and submission

 a new attribute has been added to the session to determine if the session is ongoing or not

 if it is actions related to a form are executed as usual

 if it is not a redirect to authorization endpoint is perform to follow the auth flow

 this is to avoid issues when a user is fully authenticated but use the back arraow on the browser

 this solution is not perfect as a back click before the end of the auth flow

 will not skip the page rendering

fixes AM-4342
